node/tests/swamp/sync_test.go

```diff
@@ -0,0 +1,53 @@
+package swamp
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/node"
+)
+
+/*
+Test-Case: Sync a Light Client with a Bridge Node
+Steps:
+1. Create a bridge node(BN)
+2. Start a BN
+3. Check BN is synced
+4. Create a Light Client(LC) with a trusted peer
+5. Start a LC with a defined connection to the BN
+6. Check LC is synced with BN
+*/
+func TestSyncLightWithBridge(t *testing.T) {
+	testing.Verbose()
+	sw := NewSwamp(t)
+	g := gomega.NewGomegaWithT(t)
+	bridge := sw.NewBridgeNode()
+
+	ctx := context.Background()
+
+	state := bridge.CoreClient.IsRunning()
+	require.True(t, state)
+	sw.WaitTillHeight(30)
+
+	err := bridge.Start(ctx)
+	require.NoError(t, err)
+	time.Sleep(1 * time.Second)
```

Comment:
Why 1 second?

Reply:
It's related to the IsSyncing from service.go. We need to rework the method to be a subscription so I can utilize the asyncAssert functionality correctly.